src/main/java/org/jitsi/jigasi/transcription/Participant.java

```diff
@@ -585,21 +614,49 @@ public boolean isCompleted()
      */
     private void buffer(byte[] audio)
     {
-        try
-        {
-            buffer.put(audio);
-        }
-        catch (BufferOverflowException | ReadOnlyBufferException e)
-        {
-            //e.printStackTrace();
-        }
-
-        int spaceLeft = buffer.limit() - buffer.position();
-        if(spaceLeft < EXPECTED_AUDIO_LENGTH)
-        {
-            sendRequest(buffer.array());
-            buffer.clear();
-        }
+        // note: the executorService is single-threaded and thus order
+        //       is preserved, even though there are multiple participants.
+        transcriber.executorService.submit(() ->
+           {
+               byte[] toBuffer;
+               if(silenceFilter != null)
+               {
+                   silenceFilter.giveSegment(audio);
+                   if(silenceFilter.shouldFilter())
+                   {
+                       return;
+                   }
+                   else if(silenceFilter.newSpeech())
+                   {
+                       buffer.clear();
+                       toBuffer = silenceFilter.getSpeechWindow();
+                   }
+                   else
+                   {
+                       toBuffer = audio;
+                   }
+               }
+               else
+               {
+                   toBuffer = audio;
+               }
+
+               try
+               {
+                   buffer.put(toBuffer);
+               }
+               catch (BufferOverflowException | ReadOnlyBufferException e)
+               {
+                   sendRequest(audio);
```

Comment:
Should we:
a) send this *after* sending the rest of the contents of `buffer` (line 656)?
b) use `toBuffer` instead of `audio` in case we passed through the `getSpeechWindow` branch (line 623)?

Reply:
If we catch these Exceptions I think we should:
1) `sendRequest(buffer.array())`
2) `buffer.clear()`
3) `sendRequest(toBuffer)`

Reply:
Actually, maybe the audio in the buffer is stale....

Reply:
Maybe we should clear the buffer at line 627?

Reply:
Actually the buffer is cleared when there is a new speech window (line 631) so stale audio will never be sent. So the answer stays the same.